WebDevStudios/wd_s
# Feature/refine svg workflow

### DESCRIPTION

We recently removed some `ImageMin` bits and pieces, which were causing issues with how SVGs were working (or not working) when built for production.

With that, we're no longer removing attributes like `fill` from our SVG paths.

This PR:
- Replaces all `fill` values in our `src` files with `currentColor`
- Removes `fill` and `stroke` from our SVG template tags
- Adds `color` as a value in our SVG template tags
- Adds `stroke-width` as a value in our SVG template tags

What this effectively does is, when inserting an SVG via the `_s_display_svg` template tag you can either specify `color` here _or_ specify `color` via CSS.

The nice thing about this is that it doesn't matter if your SVG is using stroke or fill – `color` will be smart enough to set either the `fill` or `stroke` value, so it eliminates a step of having to worry about which value a particular SVG is using.

The other handy thing is that, since we're using `currentColor` as the default fill on icons they should always inherit the parent color if you don't explicitly set one. So, if you have an icon inside of an `h1` and the `h1` is set to blue, the icon should automatically also be blue.

Would love to get as many folks as possible to test this, but especially @oliverharrison since he just made some updates to our SVG display template tags and @itsamoreh because the majority of the details here came from things he's found and used on recent projects.

### SCREENSHOTS

![template tag](https://d.pr/i/yCqRQf+)
![icon display markup](https://d.pr/i/q8Sw2z+)
![icon frontend display](https://d.pr/i/DU1Y4x+)

### OTHER


### STEPS TO VERIFY
1. Check out
2. Display some icons using our template tag
3. Set the `color` value in the template tag, and
4. Try to, instead, set the `color` value via CSS, and
5. Try to, also, wrap your SVG in another element and give that parent element a color to see if the color cascades down as expected

### DOCUMENTATION
Will this pull request require updating the wd_s [wiki](https://github.com/WebDevStudios/wd_s/wiki)?

Yes, we should update with notes on our workflow and usage of `currentColor` if this all looks good to everybody.
